writer.h

```diff
@@ -46,4 +79,323 @@ class voxel_writer {
 	}
 };
 
+
+class hdf_writer :public abstract_writer {
+
+public:
+	void Write(const std::string& fnc, std::string statement="default") {
+
+		abstract_chunked_voxel_storage* storage = (abstract_chunked_voxel_storage*)voxels_;
+
+		int continuous_count = 0;
+		int planar_count = 0;
+		int constant_count = 0;
+
+		hsize_t nchunks_x = storage->num_chunks().get(0);
+		hsize_t nchunks_y = storage->num_chunks().get(1);
+		hsize_t nchunks_z = storage->num_chunks().get(2);
+
+		BEGIN_LOOP(size_t(0), nchunks_x, 0U, nchunks_y, 0U, nchunks_z)
+			auto c = storage->get_chunk(ijk);
+
+		if (c == nullptr) {
```

Comment:
Can you make the whitespace formatting a bit consistent here?

The general rule (although undocumented still) for this project is `} else {` and the indentation within BEGIN_LOOP is also not consistent.